src/zenml/zen_server/deploy/terraform/recipes/aws/variables.tf

```diff
@@ -144,7 +139,7 @@ variable "create_ingress_controller" {
 }
 
 # if you already have an ingress controller, supply it's URL
-variable "ingress_controller_hostname" {
+variable "ingress_controller_ip" {
```

Comment:
do deletions / updates like this potentially break pre-existing stack recipe deployments for anyone? Do we need to support the old variables as well? i.e. how is this usually handled?

Reply:
Usually this is handled by the CLI. If somebody pulls/deploys a stack recipe, they have a local copy of the recipe files that is used for deployment. When we push an update to the repo and the user does a `zenml stack recipe pull` they get a warning saying that an existing copy exists and it would not be overwritten so their state is safe. If you want to force update you also get a warning and a prompt that confirms that you want to update the files and this could break existing deployments.

Reply:
And what about people who just use the recipes for the raw TF scripts?

Reply:
It could potentially break for people using the raw TF files if they pull a new version of the recipes to their local. It is unlikely to cause big problems because most of the time, terraform will naturally just update the deployment to match the new configuration but we don't offer guarantees.

Reply:
I guess what I'm getting at: can we deprecate these somehow? Does that even make sense?

Reply:
Ah, I get it, yes we can deprecate this if we keep the old variable intact. However, I feel that keeping both options would add to the confusion in an already complex story. Plus, on the technical side, we don't want people to continue using the hostname since it might interfere with the ingress set up through the stack recipes (assuming that the user used them to create the stack components).